mis_builder/views/mis_report_instance.xml

```diff
@@ -197,19 +197,16 @@
                                 <tree editable="bottom">
                                     <field name="sign"/>
                                     <field name="period_to_sum_id"
-                                           domain="[('report_instance_id', '=', parent.report_instance_id), ('id', '!=', parent.id)]"
                                            options="{'no_create': True, 'no_open': True}"/>
                                 </tree>
                             </field>
                             <field name="source_sumcol_accdet"
                                    attrs="{'invisible': [('source', '!=', 'sumcol')]}"/>
                             <field name="source_cmpcol_to_id"
                                    attrs="{'invisible': [('source', '!=', 'cmpcol')], 'required': [('source', '=', 'cmpcol')]}"
-                                   domain="[('report_instance_id', '=', report_instance_id), ('id', '!=', id)]"
                                    options="{'no_create': True, 'no_open': True}"/>
                             <field name="source_cmpcol_from_id"
                                    attrs="{'invisible': [('source', '!=', 'cmpcol')], 'required': [('source', '=', 'cmpcol')]}"
-                                   domain="[('report_instance_id', '=', report_instance_id), ('id', '!=', id)]"
```

Comment:
This domain is important because we can only compare columns of the same report instance. And we cannot compare a column to itself hence 'id' != id. 

Why removing it?

Reply:
It was giving nasty errors. I think that we need to resolve changing the domain dynamically in onchange methods

Reply:
Hm, I remember now. What we'd ideally need is to refer to `parent.parent` in view domains.

Yet all this did work fine in 9 and 10. Perhaps we should summon rco with an accurate description of the problem. That's not for tonight, though.

Thanks for this huge work btw. Wishing you a happy end of year.

Reply:
```
Error:
Odoo Server Error

Traceback (most recent call last):
  File "/home/odoo-11/src/odoo/odoo/fields.py", line 944, in __get__
    value = record.env.cache.get(record, self)
  File "/home/odoo-11/src/odoo/odoo/api.py", line 960, in get
    value = self._data[field][record.id][key]
KeyError: (<odoo.sql_db.Cursor object at 0x7f1f7152b550>, 1)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/odoo-11/src/odoo/odoo/http.py", line 646, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/home/odoo-11/src/odoo/odoo/http.py", line 307, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/home/odoo-11/src/odoo/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/home/odoo-11/src/odoo/odoo/http.py", line 683, in dispatch
    result = self._call_function(**self.params)
  File "/home/odoo-11/src/odoo/odoo/http.py", line 339, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/home/odoo-11/src/odoo/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/home/odoo-11/src/odoo/odoo/http.py", line 332, in checked_call
    result = self.endpoint(*a, **kw)
  File "/home/odoo-11/src/odoo/odoo/http.py", line 927, in __call__
    return self.method(*args, **kw)
  File "/home/odoo-11/src/odoo/odoo/http.py", line 512, in response_wrap
    response = f(*args, **kw)
  File "/home/odoo-11/src/odoo/addons/web/controllers/main.py", line 924, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/home/odoo-11/src/odoo/addons/web/controllers/main.py", line 916, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/home/odoo-11/src/odoo/odoo/api.py", line 687, in call_kw
    return call_kw_model(method, model, args, kwargs)
  File "/home/odoo-11/src/odoo/odoo/api.py", line 672, in call_kw_model
    result = method(recs, *args, **kwargs)
  File "/home/odoo-11/src/odoo/odoo/models.py", line 3278, in create
    record = self.browse(self._create(old_vals))
  File "/home/odoo-11/src/odoo/odoo/models.py", line 3425, in _create
    field.write(self.with_context(rel_context), vals[name], create=True)
  File "/home/odoo-11/src/odoo/odoo/fields.py", line 2273, in write
    if int(line_sudo[inverse]) != record.id:
  File "/home/odoo-11/src/odoo/odoo/models.py", line 4657, in __getitem__
    return self._fields[key].__get__(self, type(self))
  File "/home/odoo-11/src/odoo/odoo/fields.py", line 948, in __get__
    self.determine_value(record)
  File "/home/odoo-11/src/odoo/odoo/fields.py", line 1049, in determine_value
    record._prefetch_field(self)
  File "/home/odoo-11/src/odoo/odoo/models.py", line 2565, in _prefetch_field
    result = records.read([f.name for f in fs], load='_classic_write')
  File "/home/odoo-11/src/odoo/odoo/models.py", line 2509, in read
    self._read_from_database(stored, inherited)
  File "/home/odoo-11/src/odoo/odoo/models.py", line 2628, in _read_from_database
    cr.execute(query_str, params)
  File "/home/odoo-11/src/odoo/odoo/sql_db.py", line 155, in wrapper
    return f(self, *args, **kwargs)
  File "/home/odoo-11/src/odoo/odoo/sql_db.py", line 232, in execute
    res = self._obj.execute(query, params)
psycopg2.DataError: invalid input syntax for integer: "virtual_55"
LINE 1: ...OM "mis_report_kpi" WHERE "mis_report_kpi".id IN ('virtual_5...
                                                             ^
```

I'm pretty sure we are hitting an orm bug. If I read the stack trace well it's attempting to prefetch the parent record which is still a virtual_ID.